internal/api/v1beta1/framework_types.go

```diff
@@ -53,12 +53,14 @@ type FrameworkList struct {
 
 // FrameworkSpec defines the desired state of Framework
 type FrameworkSpec struct {
+	Version string `json:"version,omitempty" yaml:"version,omitempty"`
+	Name    string `json:"-" yaml:"name"`
 	// +kubebuilder:validation:MinLength=1
-	NamespaceName string `json:"namespace"`
+	NamespaceName string `json:"namespace" yaml:"namespace"`
```

Comment:
I think we can use `sigs.k8s.io/yaml` and drop `yaml` tags. The reason for this is that `sigs.k8s.io/yaml` uses json tags and this means yaml and json representations are always the same

Reply:
I made the switch to sigs.k8.io/yaml. My only concern was that this lib alphabetizes fields, so you end up with fields  like `name` and `version` near the bottom of the yaml output.

Reply:
oops, that's a valid concern!